scripts/local_gha_ci.sh

```
@@ -297,6 +309,55 @@ save_jenkins_artifacts() {
+  expected=$(extract_declared_stages "${conf_file}")
+  nstages=$(printf '%s\n' "${expected}" | awk 'NF' | wc -l | tr -d ' ')
+  if [[ "${nstages}" -lt 2 ]]; then
+    return 0
+  fi
+  if [[ ! -f "${console_log}" ]]; then
+    echo "STAGE-ORDER FAIL ${label}: console log not found (${console_log})"
+    return 1
+  fi
+  actual=$(extract_console_stages "${console_log}")
+  if [[ "${expected}" == "${actual}" ]]; then
+    echo "STAGE-ORDER PASS ${label}: $(echo ${expected} | tr '\n' ' ')"
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)